apps/jobs/views.py

```diff
@@ -2775,3 +2775,46 @@ def update_submission_started_at(request, submission_pk):
         return Response(response_data, status=status.HTTP_200_OK)
     else:
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(["PATCH"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def change_submission_status_to_cancel(request, challenge_pk, submission_pk):
+    """
+    API Endpoint for setting submission status to CANCELLED.
+    """
+    print(request.user, challenge_pk)
+    participant_team_pk = get_participant_team_id_of_user_for_a_challenge(
+        request.user, challenge_pk
+    )
+
+    try:
+        participant_team = ParticipantTeam.objects.get(pk=participant_team_pk)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {"error": "You haven't participated in the challenge"}
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    try:
+        submission = Submission.objects.get(
+            id=submission_pk,
```

Comment:
We can simply get the Submission from the `submission_pk`, why do we need participant team here?

Reply:
Addressed in above comments